source/agora/consensus/EnrollmentManager.d

```diff
@@ -544,6 +559,26 @@ public class EnrollmentManager
         return enrolls;
     }
 
+    /***************************************************************************
+
+        Get a pre-image for revelation
+
+        Params:
+            preimage = will contain the PreimageInfo if exists
+
+        Returns:
+            true if the pre-image exists
+
+    ***************************************************************************/
+
+    public bool getNextPreimage (out PreimageInfo preimage) @safe
+    {
+        auto height = this.next_reveal_height + PreimageRevealPeriod * 2;
+        if (height > ValidatorCycle - 1)
+            height = ValidatorCycle - 1;
```

Comment:
This looks completely wrong? If the block height is 1009 then we will simply not reveal the preimage?

Reply:
> If the block height is 1009 then we will simply not reveal the preimage?

As previously mentioned, the `next_reveal_height` is set based on a current block height by calling setNextRevealHeight in updateEnrolledHeight which is called in addValidatedBlock of the ledger which is being reviewed in the PR #504